elastalert/alerters/sns.py

```diff
@@ -10,9 +10,9 @@ class SnsAlerter(Alerter):
 
     def __init__(self, *args):
         super(SnsAlerter, self).__init__(*args)
-        self.sns_topic_arn = self.rule.get('sns_topic_arn', '')
-        self.sns_aws_access_key_id = self.rule.get('sns_aws_access_key_id')
-        self.sns_aws_secret_access_key = self.rule.get('sns_aws_secret_access_key')
+        self.sns_topic_arn = self.rule['sns_topic_arn']
+        self.sns_aws_access_key_id = self.rule['sns_aws_access_key_id']
```

Comment:
I don't think these are required. If the user chooses to rely on the sns_aws_profile then there should not be a need to also specify the key and secret separately.

Reply:
The body of the pull request is incorrect. I'm sorry. Only ``sns_topic_arn`` is required before modification. The contents of the correction are the correction, the addition of the test code, and the correction of the document in the writing method when the initial value is not specified.

Reply:
Revert sns_aws_access_key_id sns_aws_secret_access_key.